src/deserialize/parse.rs

```diff
@@ -397,6 +426,10 @@ fn read_map(chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Edn, Er
     loop {
         match chars.next() {
             Some((_, '}')) => return Ok(Edn::Map(Map::new(res))),
+            Some((_, ';')) => {
+                read_comment(chars)?;
+                ()
```

Comment:
You don't need `()` here (Rust will interpret a line ending in `;` as `()`.

Reply:
I do think this expression is not necessary, as it is already being considered in `parse`

Reply:
thanks for the tip on that!, I'll fix all of those similar cases

the top level parse doesn't catch inside a map, vec, set, etc., so if i comment this line the following test fails (from line 852):
```test deserialize::parse::test::parse_comment_inside_map ... FAILED```